### ISSUE TITLE
Capture type of phone number and multiple phone numbers

### ISSUE THREAD

**Issue by aliciabeckett-gov** [CDCgov/prime-simplereport#517]
To enable texting/contacting patients we need to know which phone numbers are a mobile number vs. landline or work.

- [ ] Capture type of phone number (required field)
- [ ] Allow a user to add multiple numbers
- [ ] Capture "Primary phone number" and "Additional phone numbers"

**Details**
Type of phone numbers allowed

- Mobile
- Landline

**Mock ups**
Primary phone number field
![image.png](https://images.zenhubusercontent.com/6006fd7f7815d74bc6dbed0a/1fdbd13d-6d41-4ca6-b9bc-3a719e199985)

When user clicks "Add another number", the "Additional phone number" and Phone type fields appear -->
![image.png](https://images.zenhubusercontent.com/6006fd7f7815d74bc6dbed0a/022d241a-cd32-4047-87f4-f3b29c12dec8)

**Comment by TomNUSDS** [comment]
Q: does "type" matter or only that it can accept text messages? Maybe it's better as a checkbox asking if texts are ok?

**Comment by hmyers-cms** [comment]
I wasn't originally sure what the options would be (work/home/landline/mobile/daytime/etc etc) but now that we have decided on just two, radios make more sense.

**Comment by aliciabeckett-gov** [comment]
One other thing to consider is how multiple phone numbers get passed to the data hub. @RickHawesUSDS @jimduff-usds - do you have a spec we can look at for requirements when sending multiple phone numbers?

**Comment by aliciabeckett-gov** [comment]
@maya - can we add this to the work for registering a new patient? I think it fits in that work stream better than the core app right now.

**Comment by belucid** [comment]
We need to know what to do with existing numbers.

Do we assume mobile? Assume not mobile? Or is there a 3rd state for those that represents not specified? An unspecified state for those seems best.

Seems if we move to radio buttons than it can start with neither selected for existing numbers, that just means though that we have to be sure the representation in the data can cover 3 states rather than just 2 (which means it can't be a boolean in the DB).

**Comment by belucid** [comment]
Minor UX suggestion... seeing that, the user is likely to wonder why we care, so maybe rather than just "Mobile", we make the option, "Mobile (can receive texts)". Or something to that effect.

**Comment by maya** [comment]
@aliciabeckett-gov I added it to our Product Backlog pipeline and we can chat about this at our next sprint planning. @belucid mentioned it being a dependency for our work.

**Comment by RickHawesUSDS** [comment]
@aliciabeckett-gov On your question about multiple telephone numbers. HL7 allows multiple phone numbers for a patient. It asks that each be classified as either a "home" or "business" number. Right now, the hub maps the single patient phone numbers as a "home" number, but it would nice if SR collected this information. We could add columns for secondary phone numbers as well.

**Comment by maya** [comment]
TBD until we know about text messaging services and this issue to the ice box until then.

**Comment by maya** [comment]
It seems like we have an answer on text messaging services so moving to the product backlog.

**Comment by aliciabeckett-gov** [comment]
@katiealoisi should we update this based on our learnings about contact info for K12?

**Comment by katiealoisi** [comment]
Discussed at sprint planning and decided to capture "alternate contact information" work in separate ticket

**Comment by nathancrtr** [comment]
Is the intent here that users will only be allowed to register one phone number of each type?

**Comment by katiealoisi** [comment]
No, I don't think so! Users should be allowed to register multiple phone numbers of each type if they choose.

**Comment by katiealoisi** [comment]
Can Twilio detect the type of phone number? If so, do we still need to collect the type of phone number? @aliciabeckett-gov

**Comment by aliciabeckett-gov** [comment]
@katiealoisi @nullflux and I looked into it and it can't do that unfortunately :/.

**Comment by nullflux** [comment]
@aliciabeckett-gov @katiealoisi 

When dealing with multiple phone numbers, do we care at all about the notion of a `primary` phone number? This isn't reflected in the design.

I ask because currently, the telephone number is included in the patient data that is permanently stored as part of each TestEvent. When moving to multiple numbers, there are two options for adapting this:

1. Leave the schema of permanently-stored Patient the same, persisting only *one* phone number 
2. Store all of the phone numbers in the Patient persisted on each TestEvent

When talking this through on the dev side, our original inclination with this work was #1, because changes to the shape of the permanent historic data aren't ideal (in fact we can't change the *type* of telephone; we have to remove the old field and add a new one with a different name) – but #1 raises the question **which phone number do we store in the TestEvent?**

I think that #2 is broadly *easier* to implement, but if there is a longer-term plan to add a `Make primary` button, or if the notion of `primary phone number` is important from the perspective of data modeling, I'd rather implement that now and go ahead with #1.

**TL;DR: Do we (or will we ever) care about *primary* phone number?**

**Comment by katiealoisi** [comment]
I haven't heard this directly from PHDs, but I imagine they would want to know the primary phone number so they know which number to try first when attempting to reach the patient for case investigation. What do you think @aliciabeckett-gov ?

**Comment by aliciabeckett-gov** [comment]
I actually have no idea. It would be good to ask them in any research we end up doing with PHDS.

From: Katherine Aloisi ***@***.***>
Reply-To: CDCgov/prime-simplereport ***@***.***>
Date: Wednesday, April 14, 2021 at 6:20 AM
To: CDCgov/prime-simplereport ***@***.***>
Cc: "Beckett, Alicia (CDC/OCOO/OD)" ***@***.***>, Mention ***@***.***>
Subject: Re: [CDCgov/prime-simplereport] Capture type of phone number and multiple phone numbers (#517)


I haven't heard this directly from PHDs, but I imagine they would want to know the primary phone number so they know which number to try first when attempting to reach the patient for case investigation. What do you think @aliciabeckett-gov<https://github.com/aliciabeckett-gov> ?

**Comment by katiealoisi** [comment]
I think we can get a faster and more direct answer by emailing this question to a few of our contacts. We aren't planning to do research with PHDs in this sprint and when we do, I don't think we'll be talking to the right people. Do you know of anyone involved in case investigation we can quickly ping? I'm thinking Brenna and Eli from AZ but not sure who else. Here's the actual question as I understand it: 

> If we send the public health department multiple phone numbers for one person, is it important to know which phone number is their "primary number"? Why?

**Comment by aliciabeckett-gov** [comment]
I would actually ping Kat in Pima and ask her to ask the Epis. AZ doesn’t do the case investigation, the counties do it. For that same reason, Harris County and Houston would also be good places to ask.

From: Katherine Aloisi ***@***.***>
Reply-To: CDCgov/prime-simplereport ***@***.***>
Date: Wednesday, April 14, 2021 at 10:17 AM
To: CDCgov/prime-simplereport ***@***.***>
Cc: "Beckett, Alicia (CDC/OCOO/OD)" ***@***.***>, Mention ***@***.***>
Subject: Re: [CDCgov/prime-simplereport] Capture type of phone number and multiple phone numbers (#517)


I think we can get a faster and more direct answer by emailing this question to a few of our contacts. We aren't planning to do research with PHDs in this sprint and when we do, I don't think we'll be talking to the right people. Do you know of anyone involved in case investigation we can quickly ping? I'm thinking Brenna and Eli from AZ but not sure who else. Here's the actual question as I understand it:

If we send the public health department multiple phone numbers for one person, is it important to know which phone number is their "primary number"? Why?

**Comment by aliciabeckett-gov** [comment]
For now, Primary Phone Number is a nice to have and can wait for a future iteration if it turns out to be really needed.
